azuread/data_users.go

```diff
@@ -26,7 +26,7 @@ func dataUsers() *schema.Resource {
 				Type:          schema.TypeList,
 				Optional:      true,
 				Computed:      true,
-				ConflictsWith: []string{"user_principal_names"},
+				ConflictsWith: []string{"user_principal_names", "mail_nicknames"},
```

Comment:
As a tangential change, would you mind changing the `ConflictsWith` field to `ExactlyOneOf`? This is a general improvement that would be nice to add at the same time :)